subsys/bluetooth/controller/hci/hci.c

```diff
@@ -3364,27 +3340,23 @@ void hci_acl_encode(struct node_rx_pdu *node_rx, struct net_buf *buf)
 		LL_ASSERT(0);
 		break;
 	}
-
 }
-#endif
+#endif /* CONFIG_BT_CONN */
 
 void hci_evt_encode(struct node_rx_pdu *node_rx, struct net_buf *buf)
 {
-	struct pdu_data *pdu_data;
-
-#if defined(CONFIG_BT_LL_SW_SPLIT)
-	pdu_data = (void *)node_rx->pdu;
-#else
-	pdu_data = (void *)((struct radio_pdu_node_rx *)node_rx)->pdu_data;
-#endif /* CONFIG_BT_LL_SW_SPLIT */
+	struct pdu_data *pdu_data = PDU_DATA(node_rx);
 
 	if (node_rx->hdr.type != NODE_RX_TYPE_DC_PDU) {
 		encode_control(node_rx, pdu_data, buf);
 	} else {
+#if defined(CONFIG_BT_CONN)
```

Comment:
Is it the same image if `encode_data_ctrl` is not in ifdef and the call here is under `if (IS_ENABLED(CONFIG_BT_CONN))`?

Reply:
Yes, but then you get warnings about an unused function for non-conn builds. I could replace the `#if defined` with `IS_ENABLED()` if you prefer

Reply:
I fixed this